docs/proposals/19-monitoring.md

```diff
@@ -0,0 +1,373 @@
+                action: keep
+        ```
+
+    - This configmap will be split up into 2 separate resources. One for the
+      `alerting_rules` and another for the `scrape_config`. The `alerting_rules`
+      should be converted into a `PrometheusRules` object. Since the
+      `scrape_config` only has one `job_name` we will only need one
+      `ServiceMonitor` or `PodMonitor` for this.
+
+    - Adding meta monitoring for the extensions: If the extensions need to be
+      scraped for monitoring, the extensions must bring their own [Custom
+      Resources][prom-crds].
+        - Currently the contract between extensions and gardener is that
+          anything that needs to be scraped must have the labels:
+          `prometheus.io/scrape=true` and `prometheus.io/port=<port>`. This is
+          defined [here][prom-config]. This is something that we can still
+          support with a `PodMonitor` that will scrape any pod in a specified
+          namespace with these labels.
+
+5. Add a [sidecar][grafana-sidecar] to grafana that will pickup dashboards and
+   provision them. Each dashboard gets its own configmap.
+
+   - Some of the dashboards were initially inspired by [kubernetes-mixins].
+     Gardener should reuse dashboards and alerting configs whenever possible. It
+     is not clear yet how to integrate this into Gardener.
+
+6. Migrating to the new monitoring stack:
+    1. Deploy the [prometheus-operator] and its custom resources.
+    1. Add a feature gate to optionally deploy the custom resources defined in the Proposal.
+    1. Add a feature gate to optionally disable the current monitoring stack.
+    1. Co-deploy both monitoring stacks.
+    1. Ensure that both stacks are equivalent.
+    1. Switch over to the monitoring model defined in this GEP.
+    1. Deprecate the current monitoring stack.
+    1. Delete the current monitoring stack.
+
+## Alternatives
+
+1. Continue using the current setup.
+
+[additional scrape config]: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/additional-scrape-config.md
+[agent mode]: https://prometheus.io/blog/2021/11/16/agent/
+[alerting-doc]: https://github.com/gardener/gardener/blob/master/docs/monitoring/alerting.md#alerting-for-users
+[API]: https://github.com/prometheus-operator/prometheus-operator/tree/main/pkg/apis/monitoring
+[apiserver-example]: https://github.com/gardener/gardener/blob/0f4d22270927e2aee8b821f858fb76162ccd8a86/charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml#L311
+[extension-contract]: https://github.com/gardener/gardener/blob/master/docs/extensions/logging-and-monitoring.md
+[gardener-resource-manager]: https://github.com/gardener/gardener/blob/eec37223cb90475ec3e023136a7d5ba28ad48f0d/pkg/operation/botanist/component/resourcemanager/monitoring.go
+[grafana-sidecar]: kiwigrid/k8s-sidecar
+[kube-prometheus]: https://github.com/prometheus-operator/kube-prometheus
```

Comment:
That is a valid concern. We could use older versions that support older K8s versions. We would need to manage this though. This idea of using kube-prometheus/kubernetes-mixins is still in the exploratory phase, so maybe it does not make sense to use. However, I do think if it is possible it would be a big win for us.